src/Lumi/Components/Select/SelectBackend.purs

```diff
@@ -201,12 +203,14 @@ selectBackend = make component
               isOptionSelected =
                 Array.any (\v -> (props.toSelectOption v).value == valueToAdd) props.value
             when (not isOptionSelected) do
-              self.props.onChange
-                if self.props.allowMultiple
-                  then self.props.value <> [ option ]
+              props.onChange
+                if props.allowMultiple
+                  then props.value <> [ option ]
                   else [ option ]
-            send self $ SetSearchTerm ""
-            send self CloseSelect
+            self.setState _
+              { isOpen = false
+              , searchTerm = ""
+              }
```

Comment:
Both of these effects run synchronously before the 2nd has a chance to see a new state value. Just manipulating state directly instead of deferring to `CloseSelect` avoids that race condition at the cost of some duplicate logic.